cfd_protocol/tests/cfds.rs

```diff
@@ -625,33 +680,50 @@ fn check_cfd_txs(
 
     // CETs:
 
-    for (tx, _, digits) in maker_cfd_txs.cets.clone().into_iter() {
-        let price = gen_price(rng, &digits);
-        let oracle_sigs = oracle.attest_price(price, &event.nonces);
-
-        build_and_check_cet(
-            tx,
-            &taker_cfd_txs.cets,
-            (&maker_sk, &maker_pk),
-            &taker_pk,
-            (price, &oracle_sigs),
-            (&signed_commit_tx_maker, &commit_desc, commit_amount),
-        )
-        .expect("valid maker cet");
+    for maker_cets_by_event in maker_cfd_txs.cets.clone().into_iter() {
```

Comment:
You seem to be cloning individual items as part of the for-loop as well. Just `.iter()` would be more idiomatic then.

Reply:
But then I would need more clones further down.